barretenberg/bbup/shell.ts

```diff
@@ -0,0 +1,83 @@
+import { execSync } from "child_process";
+import logSymbols from "log-symbols";
+import { Ora } from "ora";
+import os from "os";
+import axios from "axios";
+import fs from "fs";
+import { createGunzip } from "zlib";
+import tar from "tar-fs";
+import { promisify } from "util";
+
+import { pipeline } from "stream";
+import path from "path";
+
+export function sourceShellConfig() {
+  const shell = execSync("echo $SHELL", { encoding: "utf-8" }).trim();
+
+  if (shell.includes("bash")) {
+    process.env.PATH = execSync("echo $PATH", { encoding: "utf-8" }).trim();
+  } else if (shell.includes("zsh")) {
+    process.env.PATH = execSync('zsh -c "echo $PATH"', {
+      encoding: "utf-8",
+    }).trim();
+  } else if (shell.includes("fish")) {
+    process.env.PATH = execSync('fish -c "echo $PATH"', {
+      encoding: "utf-8",
+    }).trim();
+  }
+}
+
+export function exec(cmd: string, options = {}) {
+  return execSync(cmd, {
+    encoding: "utf-8",
+    stdio: "pipe",
+    ...options,
+  });
+}
+export async function installBB(version: string, spinner: Ora) {
+  let architecture = os.arch();
+  if (architecture === "arm64") {
+    architecture = "aarch64";
+  } else if (architecture === "x64") {
+    architecture = "x86_64";
+  }
+
+  let platform: string = os.platform();
+  if (platform === "darwin") {
+    platform = "apple-darwin";
+  } else if (platform === "linux") {
+    platform = "linux-gnu";
+  }
+
+  const home = os.homedir();
+  const bbPath = path.join(home, ".bb");
+
+  spinner.start(`Installing to ${bbPath}`);
+  const tempTarPath = path.join(home, "temp.tar.gz");
```

Comment:
Making a utility for a temp directory is more sound. Just takes two tools that think this is a good path to run at the same time to cause very hard to debug failures

Reply:
Sorry, using a utility, should come with node